python/dronin/telemetry.py

```diff
@@ -479,6 +479,137 @@ def _do_io(self, finish_time):
 
         return did_stuff
 
+class HIDTelemetry(BidirTelemetry):
+    """ HID telemetry interface """
+    def __init__(self, vidpid=None, vid=None, pid=None, speed=115200, *args, **kwargs):
+        """ Creates telemetry instance talking over USB HID.  Likely to only
+        work on linux for now.
+
+         - vidpid: a string, of two hex numbers separated by colon,
+         like 39ee:4a3e, that in turn specifies the vendor id and product id
+         to talk to.
+
+        Meaningful parameters passed up to TelemetryBase include: githash,
+        service_in_iter, iter_blocks, use_walltime
+
+        Requires the module pyusb
+        """
+
+        if vid is None:
+            try:
+                raw = str.split(vidpid, ':')
+
+                if len(raw) != 2:
+                    raise
+
+                vid = int(raw[0], base=16)
+                pid = int(raw[1], base=16)
+            except Exception:
+                raise ValueError("Invalid value of vidpid")
+
+        import usb.core
+        import usb.util
+
+        dev = usb.core.find(idVendor = vid, idProduct = pid)
+
+        if dev is None:
+            raise RuntimeError("No fc found")
+
+        # XXX do better things here
+        if dev.is_kernel_driver_active(0):
+            dev.detach_kernel_driver(0)
+        if dev.is_kernel_driver_active(1):
+            dev.detach_kernel_driver(1)
+        if dev.is_kernel_driver_active(2):
+            dev.detach_kernel_driver(2)
+
+        #dev.set_configuration()
```

Comment:
So, we're supposed to tell the device to take on a configuration.  Our stuff only supports one.

If we do tell it to change configuration, the OS often re-enumerates it, and then seizes it, which is annoying.